server/pairing/client.go

```diff
@@ -71,13 +92,19 @@ func NewBaseClient(c *ConnectionParams, logger *zap.Logger) (*BaseClient, error)
 	var serverCert *x509.Certificate
 	var certErrs error
 
+	netIPs, err := server.FindReachableAddressesForPairingClient(c.netIPs)
+	if err != nil {
+		logger.Error("[local pair client] failed to find reachable addresses", zap.Error(err), zap.Any("netIPs", netIPs))
+		signal.SendLocalPairingEvent(Event{Type: EventConnectionError, Error: err.Error(), Action: ActionConnect})
+		return nil, err
+	}
+
 	maxRetries := 3
 	for i := 0; i < maxRetries; i++ {
-		baseAddress, serverCert, certErrs = findServerCert(c)
+		baseAddress, serverCert, certErrs = findServerCert(c, netIPs)
```

Comment:
Is it possible for the search for a server certificate to last longer?
Like, if we want to check 5 addresses, 4 of them return err and the fifth needs time. Is that scenario possible at all?

Reply:
each address may last up to 1 second